x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/context.ts

```diff
@@ -14,18 +13,103 @@ import { MessageAddEvent } from '../../common/conversation_complete';
 import { FunctionVisibility } from '../../common/functions/types';
 import { MessageRole } from '../../common/types';
 import { createFunctionResponseMessage } from '../../common/utils/create_function_response_message';
+import {
+  KnowledgeBaseQueryContainer,
+  KnowledgeBaseSource,
+  KnowledgeBaseStatus,
+} from '../service/knowledge_base_service/types';
 import { recallAndScore } from '../utils/recall/recall_and_score';
 
 const MAX_TOKEN_COUNT_FOR_DATA_ON_SCREEN = 1000;
 
 export const CONTEXT_FUNCTION_NAME = 'context';
 
+interface ContextToolResponseError {
+  content: string;
+  data: {
+    error: {
+      message: string;
+    };
+  };
+}
+
+interface ScreenContent {
+  data_on_screen?: Array<{ name: string; description: string; value: unknown }> | undefined;
+  screen_description: string;
+}
+
+export interface ContextToolResponseV1 {
+  content: ScreenContent & {
+    learnings: Array<{
+      text: string;
+      score: string;
+      id: string;
+    }>;
+  };
+  data: {
+    scores: Array<{
+      id: string;
+      score: number;
+    }>;
+    suggestions: Array<{
+      text: string;
+      score: number;
+      id: string;
+    }>;
+  };
+}
+
+interface ContextEntryFormatted {
+  id: string;
+  title: string;
+  document: unknown;
+  relevanceScore: number | null;
+  llmScore: number | null;
+}
+
+export interface ContextEntry {
+  id: string;
+  title: string;
+  text: string;
+  document: unknown;
+  source: KnowledgeBaseSource;
+  score: number | null;
+  llmScore: number | null;
+  selected: boolean;
+  truncated?: {
+    truncatedText: string;
+    truncatedTokenCount: number;
+    originalTokenCount: number;
+  };
+}
+
+export interface ContextToolResponseV2 {
+  content: ScreenContent & {
+    entries: ContextEntryFormatted[];
+  };
+  data: {
+    entries: ContextEntry[];
+    queries: KnowledgeBaseQueryContainer[];
+  };
+}
+
+export type ContextToolResponse =
+  | ContextToolResponseV1
+  | ContextToolResponseV2
```

Comment:
Why does this need versioning?

Reply:
Because we have existing conversations with a different response schema, so we need to take this into account when rendering